### PR TITLE
[Boost] Fix Critical CSS error display, combine classes

### PR DESCRIPTION
**Pull request by thingalon**
Companion PR: https://github.com/Automattic/boost-cloud/pull/228

When generating Cloud CSS failed with a showstopper, they were sometimes not displayed in the UI correctly. Additionally, individual errors were not shown; they were being hidden in the UI.

This PR fixes those issues, and cleans up duplicate code.

## Proposed changes:
- Fixes some display bugs with Cloud CSS. When failures occurred, they were not being displayed properly.
- Simplifies our codebase at the same time, removing the separate handling for Cloud CSS and Critical CSS status objects.

### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* Update your local Cloud CSS service with https://github.com/Automattic/boost-cloud/pull/228
* Test Generating Critical CSS locally - both with and without errors, ensure it works.
* Test Generating Cloud CSS - both with and without errors.

### CODE REVIEW

File: projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php [hunk 1]
```diff
@@ -22,7 +45,57 @@ public function request_methods() {
...
+			}
+
+			// Collate URL errors.
+			$urls   = empty( $data['urls'] ) || ! is_array( $data['urls'] ) ? array() : $data['urls'];
+			$errors = array_map(
+				function ( $url ) use ( $urls ) {
+					return array_merge( array( 'url' => $url ), $urls[ $url ] );
+				},
+				array_keys( $data['urls'] )
+			);
```
Comment (pyronaur) [inline projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php]: This is a bit confusing, maybe just use `foreach` instead? There's also URLs array above that's only partially used in the anonymous function.

File: projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php [hunk 2]
```diff
@@ -22,7 +45,57 @@ public function request_methods() {
...
+
+			$state->set_provider_errors( $provider_key, $errors );
+		}
+
+		// Save the state changes.
+		$state->save();
+
+		return $api_successful;
```
Comment (pyronaur) [inline projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php:94]: This function always returns success - I'm not seeing any negative paths, is that correct?

**Review by pyronaur — CHANGES_REQUESTED** [review]
More red than green! 🎉

Left two small comments.

File: projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php (hunk 1 above)
Comment (thingalon) [reply projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php]: Sure; simplified :)

File: projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php (hunk 2 above)
Comment (thingalon) [reply projects/plugins/boost/app/rest-api/endpoints/Update_Cloud_CSS.php:94]: No; there are return paths that return a WP_Error if the request is bad.

**Review by pyronaur — APPROVED** [review]

**Pull request merged**